packages/terra-date-input/src/DateInput.jsx

```diff
@@ -30,6 +31,11 @@ const propTypes = {
    * Overrides default locale display format.
    */
   displayFormat: PropTypes.oneOf(['month-day-year', 'day-month-year']),
+  /**
+   * @private
+   * To check if help element is provided by the field or not.
```

Comment:
```suggestion
   * NOTICE: Internal prop to be used only by Terra framework. This component provides a built-in format mask that is required to be displayed to users for proper accessibility and must not be removed. 'DateInputField' is permitted to set this prop because it provides the same format mask in its 'help' prop.
```

Suggestion to switch propType to a boolean, default to false, and update prop name (e.g. `useExternalFormatMask`)

Reply:
Updated here https://github.com/cerner/terra-framework/pull/1230/commits/4690dd459dd9b4a4df83b92a773d5bf707d36d94.